pkg/codegen/python/gen.go

```diff
@@ -2766,6 +2783,22 @@ func generateModuleContextMap(tool string, pkg *schema.Package, info PackageInfo
 		mod.extraSourceFiles = append(mod.extraSourceFiles, p)
 	}
 
+	// Setup modLocator so that mod.typeDetails finds the right
+	// modContext for every ObjectType.
+	modLocator := &modLocator{
```

Comment:
I don't understand why we need the extra `struct` here instead of just passing a handle to the `func`

Reply:
I was thinking in case it needs more methods later like EnumType, and I also like to have my `obj.method()` syntax to have an idea what the call is doing, but yeah technically could just have been func.